gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsInfoApiMonitor.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.clustermonitor;
+
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpClientConfig;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.jetty.JettyHttpClient;
+import io.trino.client.ServerInfo;
+import io.trino.gateway.ha.config.ProxyBackendConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+
+import static io.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static io.airlift.http.client.JsonResponseHandler.createJsonResponseHandler;
+import static io.airlift.http.client.Request.Builder.prepareGet;
+import static io.airlift.json.JsonCodec.jsonCodec;
+
+public class ClusterStatsInfoApiMonitor
+        implements ClusterStatsMonitor
+{
+    //TODO: make client options configurable
+    private static final HttpClient client = new JettyHttpClient(new HttpClientConfig());
+    private static final Logger log = LoggerFactory.getLogger(ClusterStatsInfoApiMonitor.class);
+
+    @Override
+    public ClusterStats monitor(ProxyBackendConfiguration backend)
+    {
+        return ClusterStats.builder(backend.getName()).healthy(isReadyStatus(backend.getProxyTo()))
+                .proxyTo(backend.getProxyTo())
+                .externalUrl(backend.getExternalUrl())
+                .routingGroup(backend.getRoutingGroup()).build();
+    }
+
+    private boolean isReadyStatus(String baseUrl)
+    {
+        Request request = prepareGet()
+                .setUri(uriBuilderFrom(URI.create(baseUrl)).appendPath("/v1/info").build())
+                .build();
+
+        try {
+            ServerInfo serverInfo = client.execute(request, createJsonResponseHandler(jsonCodec(ServerInfo.class)));
```

Comment:
Extract jsoncoded and use it.
```java
    private static final JsonCodec<ServerInfo> SERVER_INFO_JSON_CODEC = jsonCodec(ServerInfo.class);
...
            ServerInfo serverInfo = client.execute(request, createJsonResponseHandler(SERVER_INFO_JSON_CODEC));
```

Reply:
Thanks for clarifying, I misunderstood the previous comment

Reply:
I extracted the entire response handler btw, let me know if you would prefer to just have the codec as a static. It doesn't appear to me that there is any state that would prevent sharing a single `ResponseHandler` across threads